From b841cbfc3b5df8b14a33615b4bcb5402f2c59fb5 Mon Sep 17 00:00:00 2001
From: thrishalig <thrishalig@ami.com>
Date: Thu, 12 Sep 2024 17:40:28 +0530
Subject: [PATCH] OT-0.2 LF and Intel sync patch recreation dbus-sensors Due to
 Intel and LF sync 2.0 patch is failing ,Recreated below patch
 0001-converted-index-to-0-based-and-made-pwm-starts-from-.patch

converted index to 0 based and made pwm starts from 0

Signed-off-by: thrishalig <thrishalig@ami.com>
---
 src/FanMain.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/FanMain.cpp b/src/FanMain.cpp
index 96cd4be..ba2893a 100644
--- a/src/FanMain.cpp
+++ b/src/FanMain.cpp
@@ -189,7 +189,7 @@ bool findPwmPath(const fs::path& directory, unsigned int pwm, fs::path& pwmPath)
     std::error_code ec;
 
     /* Assuming PWM file is appeared in the same directory as fanX_input */
-    auto path = directory / ("pwm" + std::to_string(pwm + 1));
+    auto path = directory / ("pwm" + std::to_string(pwm));
     bool exists = fs::exists(path, ec);
 
     if (ec || !exists)
@@ -323,7 +323,7 @@ void createSensors(
                 std::string cfgIntf = configInterfaceName(sensorTypes[fanType]);
 
                 // convert to 0 based
-                size_t index = std::stoul(indexStr) - 1;
+                size_t index = std::stoul(indexStr);
 
                 const char* baseType = nullptr;
                 const SensorData* sensorData = nullptr;
@@ -516,7 +516,7 @@ void createSensors(
                         }
 
                         fs::path pwmEnableFile =
-                            "pwm" + std::to_string(pwm + 1) + "_enable";
+                            "pwm" + std::to_string(pwm) + "_enable";
                         fs::path enablePath =
                             pwmPath.parent_path() / pwmEnableFile;
                         enablePwm(enablePath);
@@ -531,7 +531,7 @@ void createSensors(
                         }
                         else
                         {
-                            pwmName = "Pwm_" + std::to_string(pwm + 1);
+                            pwmName = "Pwm_" + std::to_string(pwm);
                         }
 
                         // Check PWM sensor mutability
-- 
2.25.1

